docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,861 @@
+# Installing OpenShift on OpenStack User-Provisioned Infrastructure
+
+The User-Provisioned Infrastructure (UPI) process installs OpenShift in stages, providing opportunities for modifications or integrating with existing infrastructure.
+
+It contrasts with the fully-automated Installer-Provisioned Infrastructure (IPI) which creates everything in one go.
+
+With UPI creating the cloud (OpenStack) resources (e.g. Nova servers, Neutron ports, security groups) is the responsibility of the person deploying OpenShift.
+
+The installer is still used to generate the ignition files and monitor the installation process.
+
+This provides a greater flexibility at the cost of a more explicit and interactive process.
+
+
+## Prerequisites
+
+The requirements for UPI are broadly similar to the [ones for OpenStack IPI][ipi-reqs]:
+
+[ipi-reqs]: ./openstack#openstack-requirements
+
+- OpenStack account with `clouds.yaml`
+  - This document will use a cloud called `openstack`
+- Nova flavors
+  - This document will use `m1.xlarge` for masters and `m1.large` for workers
+- An external subnet you want to use for floating IP addresses
+  - This document will use `external`
+- The `openshift-install` binary in your `$PATH`
+- The [Red Hat CoreOS][rhcos] image in Glance
+  - This document will use the `rhcos` image name
+- A subnet range for the Nova servers / OpenShift Nodes
+  - This range must not conflict with your existing network
+  - This document will use `192.0.2.0/24`
+- A cluster name you will want to use
+  - This document will use `openshift`
+- A base domain
+  - This document will use `example.com`
+- OpenShift Pull Secret
+- DNS infrastructure you can configure (you'll need to add two records there)
+
+[rhcos]: https://www.openshift.com/learn/coreos/
```

Comment:
Did you avoid providing a deeper link on purpose?

Reply:
I couldn't really find a better one. If you've got suggestions, I'm all ears!

Reply:
> If you've got suggestions

I really haven't! And that's precisely the point: the page reads "RHEL CoreOS is available as part of Red Hat OpenShift", which is a bit of a chicken and egg problem... :thinking:

Reply:
If I even was to find a direct link, I would even not know what are our policies regarding deeplinking Red Hat products.

Reply:
_I have nothing useful to add: please consider as resolved_

Reply:
Ok, will do. We can always improve this later.